File: backend/src/common/sc-ids.ts
/**
 * Нормализует sc_track_id к чистому числовому виду.
 * Клиент может прислать URN ("soundcloud:tracks:12345") или число ("12345").
 * Qdrant хранит point_id как int, `indexed_tracks.scTrackId` — числовая строка.
 * Возвращает null если вход невалиден — вызывающий должен игнорировать событие.
 */
export function normalizeScTrackId(input: string | null | undefined): string | null {
  if (!input) return null;
  const last = input.includes(':') ? input.split(':').pop() ?? '' : input;
  return /^\d+$/.test(last) ? last : null;
}

File: backend/src/common/sc_ids.rs
pub fn normalize_sc_track_id(input: &str) -> Option<String> {
    if input.is_empty() {
        return None;
    }
    let last = if input.contains(':') {
        input.rsplit(':').next().unwrap_or("")
    } else {
        input
    };
    if !last.is_empty() && last.bytes().all(|b| b.is_ascii_digit()) {
        Some(last.to_string())
    } else {
        None
    }
}

/// "soundcloud:tracks:1234" → "1234". Без валидации формата — для случаев
/// когда URN заведомо корректен (приходит из SC ответа / роута).
pub fn extract_sc_id(urn: &str) -> &str {
    urn.rsplit_once(':').map(|(_, id)| id).unwrap_or(urn)
}
